### ISSUE TITLE
Multi Class - Running Loss nan - NaN or Inf found in input tensor

### ISSUE THREAD

**Issue by hammad26** [ThilinaRajapakse/simpletransformers#100]
Hi
I am using Multi Class, xlm model. I have converted labels from 0 to n too. At the time of training I am getting following issue:

![image](https://user-images.githubusercontent.com/12643784/71153829-f1a96400-225b-11ea-8c5e-c46561c946ba.png)

Although I have checked my data. It is looking fine. Can you guide in it? Thanks

**Comment by hammad26** [comment]
More info about data:
Data Size: 5000
Labels: 3

On given example in ReadMe it is running successfully. Is it about large data set that this issue is coming?

**Comment by ThilinaRajapakse** [comment]
5000 samples isn't really a large dataset. Try using `'fp16_opt_level': 'O1'`.

**Comment by hammad26** [comment]
I am getting Kernal restarting issue and solved it using `'fp16_opt_level': 'O3'` as mentioned there. #99

**Comment by ThilinaRajapakse** [comment]
Try turning off fp16 as it sounds like Apex might be causing issues.

**Comment by hammad26** [comment]
Thanks. Its working perfectly fine now.

**Comment by salil-gtm** [comment]
I'm facing the same `Running loss: nan` issue on a regression task. I'm tried using 'fp16': False in args as mentioned above, but the issue persists.

**Comment by ThilinaRajapakse** [comment]
Please share a script which shows the training arguments you are using.

**Comment by salil-gtm** [comment]
```
model = ClassificationModel('roberta', 'distilroberta-base',num_labels=1, args={"reprocess_input_data": True, "overwrite_output_dir": True,
    'no_cache': True, 'fp16': False,  'use_early_stopping': True,
    'early_stopping_patience': 3,
    'manual_seed': 4,
    'regression': True,                                     
    'max_seq_length': 142,                                                    
    'num_train_epochs': 1,
    'train_batch_size': 128,
    'eval_batch_size': 64,
    'gradient_accumulation_steps': 1,
    'learning_rate': 5e-5,
    'save_steps': 50000,"save_model_every_epoch": False})

# Train the model
model.train_model(train_df)
```

**Comment by salil-gtm** [comment]
> ```
> model = ClassificationModel('roberta', 'distilroberta-base',num_labels=1, args={"reprocess_input_data": True, "overwrite_output_dir": True,
>     'no_cache': True, 'fp16': False,  'use_early_stopping': True,
>     'early_stopping_patience': 3,
>     'manual_seed': 4,
>     'regression': True,                                     
>     'max_seq_length': 142,                                                    
>     'num_train_epochs': 1,
>     'train_batch_size': 128,
>     'eval_batch_size': 64,
>     'gradient_accumulation_steps': 1,
>     'learning_rate': 5e-5,
>     'save_steps': 50000,"save_model_every_epoch": False})
> 
> # Train the model
> model.train_model(train_df)
> ```

Everything working fine ! Error was due to NaN in the training text data.
